dhis-2/dhis-services/dhis-service-field-filtering/src/main/java/org/hisp/dhis/fieldfiltering/FieldPathHelper.java

```diff
@@ -295,6 +291,35 @@ private void applyExclusions(List<FieldPath> exclusions, Map<String, FieldPath>
   // Helpers
   // ----------------------------------------------------------------------------------------------------------------
 
+  /**
+   * Method that checks whether a field is equal to the root field. <br>
+   * examples: <br>
+   *
+   * <ul>
+   *   <li>fullFieldPath = "root", field = "root" -> return true
+   *   <li>fullFieldPath = "root.name", field = "root" -> return true
+   *   <li>fullFieldPath = "root.name", field = "name" -> return false
+   *   <li>fullFieldPath = "username", field = "user" -> return false
+   * </ul>
+   *
+   * @param fullFieldPath field path that represents the full path of a given field(in dot
+   *     notation). It might look like any of the following examples: <br>
+   *     <ul>
+   *       <li>root
+   *       <li>root.name
+   *       <li>root.name.last
+   *     </ul>
+   *
+   * @param field is the actual field which should be checked. This will only be 1 word (no dot
+   *     notation)
+   * @return true if the field is equal to the fullFieldPath root <br>
+   */
+  public static boolean fieldEqualsRootField(String fullFieldPath, String field) {
+    if (ObjectUtils.anyNull(fullFieldPath, field)) return false;
+    String root = fullFieldPath.split("\\.")[0];
```

Comment:
That seems a unnecessary complicated way to check this. `fullFieldPath.substring(0, fullFieldPath.indexOf('.'))` should give `root` I think.

Reply:
the beauty of `fullFieldPath.split("\\.")[0]` is that it handles the lack of any `.` in the string.
whereas `fullFieldPath.indexOf('.')` could return -1 & then we're into using more logic to get the result we want.